### PR TITLE
[zfsboot] Fix boot env back compat

### PR DESCRIPTION
**Pull request by ghost**
Preliminary fix while we figure out a long term solution upstream.

zfsboot must try zfsloader before loader in order to remain compatible
with boot environments created prior to zfs functionality being rolled
into loader proper.

Ticket: #80094

### CODE REVIEW

**Review by kithrup — APPROVED** [review]
This is ok as it is; the only _possible_ change I might make -- and this isn't a requirement, just something worth considering -- would be to make the set of loader paths an array, and cycle through them until one works.  However, that would mean using a strcpy/strlen function, or an array of <str, len> pairs, which also complicates it.

**Comment by ghost** [comment]
I thought of having a configurable list of things to try, perhaps set by zfsbootcfg(8). It would have to be some property of the pool itself, because any file-based config (i.e. /boot.config) or per-dataset properties must be assumed to not already exist if this mechanism is to be backward compatible.

A static list of Pascal-style strings as you suggest is simple by comparison, and I think it is a good improvement on what I originally had.

**Comment by ghost** [comment]
Confirmed working on fn03.

File: stand/i386/zfsboot/zfsboot.c [hunk 1]
```diff
@@ -839,15 +848,16 @@ main(void)
 	reboot();
 
     /*
-     * Try to exec /boot/loader. If interrupted by a keypress,
-     * or in case of failure, try to load a kernel directly instead.
+     * Try to exec a loader or the kernel directly.
+     * If interrupted by a keypress, or in case of failure,
+     * drop the user to a boot2 prompt.
```
Comment (kithrup) [inline stand/i386/zfsboot/zfsboot.c]: Change the comment to match the new code.  I'd suggest something like, "Loop through the list of loaders/kernel" and merge that in with the language of the existing comment.

Comment (ghost) [reply stand/i386/zfsboot/zfsboot.c]: Added commit to improve comments.

**Pull request merged**